backend/src/main/java/com/bakdata/conquery/apiv1/query/CQElement.java

```diff
@@ -73,14 +73,8 @@ public void collectRequiredQueries(Set<ManagedExecution<?>> requiredQueries) {}
 		this.collectRequiredQueries(set);
 		return set;
 	}
-
-	public ResultInfoCollector collectResultInfos(PrintSettings cfg) {
-		ResultInfoCollector collector = new ResultInfoCollector();
-		collectResultInfos(collector);
-		return collector;
-	}
 	
-	public abstract void collectResultInfos(ResultInfoCollector collector);
+	public abstract void collectResultInfos(List<ResultInfo> collector);
```

Comment:
Wäre hier consumer nicht sogar besser? In der Api Interessiert mich ja nicht wo ich meine Sachen hin schicke

Reply:
kann leider nur entweder `add` oder `addAll`, nicht beides.

Reply:
Ist es wichtig addAll an der Stelle zu haben? Es macht ja meistens auch nicht arg viel mehr als forEach(add), und ggf bei einer arraylist zu reservieren. Aber performance auf dem niveau hier ja nicht das wichtigste?